docs/CNDP_README.md

```diff
@@ -0,0 +1,323 @@
+
+### Step 3: Run UPF in System 1
+
+1) Setup hugepages in the system 1. For example, to setup 8GB of total huge pages (8 pages each of size 1GB in each NUMA node) using DPDK script, use the command `dpdk-hugepages.py -p 1G --setup 8G`
+
+2) Enable cndp mode, use appropriate netdev interface and uncomment jsonc config file in configuration file  [upf.json](./conf/upf.json)
+
+```
+diff --git a/conf/upf.json b/conf/upf.json
+index 62fa435..169dc5a 100644
+--- a/conf/upf.json
++++ b/conf/upf.json
+@@ -3,8 +3,8 @@
+     "": "mode: af_xdp",
+     "": "mode: af_packet",
+     "": "mode: sim",
+-    "": "mode: cndp",
+-    "mode": "dpdk",
++    "": "mode: dpdk",
++    "mode": "cndp",
+
+     "table_sizes": {
+         "": "Example sizes based on sim mode and 50K sessions. Customize as per your control plane",
+@@ -62,14 +62,14 @@
+
+     "": "Gateway interfaces",
+     "access": {
+-        "ifname": "ens803f2",
+-        "": "cndp_jsonc_file: /opt/bess/bessctl/conf/cndp_upf_1worker.jsonc"
++        "ifname": "enp134s0",
++        "cndp_jsonc_file": "/opt/bess/bessctl/conf/cndp_upf_1worker.jsonc"
+     },
+
+     "": "UE IP Natting. Update the line below to `\"ip_masquerade\": \"<ip> [or <ip>]\"` to enable",
+     "core": {
+-        "ifname": "ens803f3",
+-        "": "cndp_jsonc_file: /opt/bess/bessctl/conf/cndp_upf_1worker.jsonc",
++        "ifname": "enp136s0",
++        "cndp_jsonc_file": "/opt/bess/bessctl/conf/cndp_upf_1worker.jsonc",
+         "": "ip_masquerade: 18.0.0.1 or 18.0.0.2 or 18.0.0.3"
+     },
+```
+
+3) Enable cndp mode in script file  [docker_setup.sh](./scripts/docker_setup.sh)
+
+```
+diff --git a/scripts/docker_setup.sh b/scripts/docker_setup.sh
+index 7aff6a6..1a8e2fd 100755
+--- a/scripts/docker_setup.sh
++++ b/scripts/docker_setup.sh
+@@ -15,8 +15,8 @@ metrics_port=8080
+ # "af_packet" uses AF_PACKET sockets via DPDK's vdev for pkt I/O.
+ # "sim" uses Source() modules to simulate traffic generation
+ # "cndp" use kernel AF-XDP. It supports ZC and XDP offload if driver and NIC supports it. It's tested on Intel 800 series n/w adapter.
+-mode="dpdk"
+-#mode="cndp"
++#mode="dpdk"
++mode="cndp"
+ #mode="af_xdp"
+ #mode="af_packet"
+ #mode="sim"
+```
+
+4) Modify [cndp_upf_1worker.jsonc](./conf/cndp_upf_1worker.jsonc) file lports section to update the access and core netdev interface name and required queue id.
```

Comment:
Additionally, it would be good to show a `diff` file about the changes needed in `cndp_upf_1worker.jsonc`

Reply:
ok